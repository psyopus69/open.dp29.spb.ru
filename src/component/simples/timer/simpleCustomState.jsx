import React, {Fragment} from "react";
import {PropTypes} from "prop-types";

// Simple for HOC withTimer
const SimpleCustomState = (props) => {
  const {getStateCustom, setStateCustom} = props;
  return <Fragment>
    <div>Test 1.4 SimpleCustomState React</div>
    <button onClick={() => {
      // eslint-disable-next-line no-console
      console.log(getStateCustom());
    }}>console status</button>
    <div>setCustomState <input onChange={(event) => {
      // console.log(event.target.value);
      setStateCustom(event.target.value, (state) => {
        // eslint-disable-next-line no-console
        console.log(`state after change -> ${state}`);
      });
    }}/></div>
  </Fragment>;
};
SimpleCustomState.propTypes = {
  getStateCustom: PropTypes.func.isRequired,
  setStateCustom: PropTypes.func.isRequired,
};

export {
  SimpleCustomState,
};
