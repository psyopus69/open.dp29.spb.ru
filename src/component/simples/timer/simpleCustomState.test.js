import React from "react";
import renderer from "react-test-renderer";
import {SimpleCustomState} from "./simpleCustomState";

it(`panel with setStatus & getStatus`, () => {
  const tree = renderer.create(<SimpleCustomState getStateCustom={jest.fn()} setStateCustom={jest.fn()}
  />);
  expect(tree).toMatchSnapshot();
});
