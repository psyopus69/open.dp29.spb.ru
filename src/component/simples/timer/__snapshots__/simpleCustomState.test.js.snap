// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`panel with setStatus & getStatus 1`] = `
Array [
  <div>
    Test 1.4 SimpleCustomState React
  </div>,
  <button
    onClick={[Function]}
  >
    console status
  </button>,
  <div>
    setCustomState 
    <input
      onChange={[Function]}
    />
  </div>,
]
`;
